setup.py

```diff
@@ -271,85 +270,28 @@ def run(self):
     history = history_file.read()
 
 
-requirements = [
-    "certifi==2021.5.30",
-    "attrs==21.2.0",
-    "click==8.0.3",
-    "msgpack==0.6.0",
-    "wsproto==1.0.0",
-    "h11==0.12.0",
-    # Can use marshmallow or the toasted flavour as you like ;-)
-    # "marshmallow==2.14.0",
-    "toastedmarshmallow==0.2.6",
-    "pendulum==2.1.2",
-    "PyNaCl==1.4.0",
-    "trio==0.18.0",
-    "trio_typing==0.5.0",
-    "async_generator>=1.9",
-    'contextvars==2.1;python_version<"3.7"',
-    'typing-extensions==3.10.0.2;python_version<"3.8"',
-    "sentry-sdk==1.3.1",
-    "structlog==21.1.0",
-    "importlib_resources==1.0.2",
-    "colorama==0.4.0",  # structlog colored output
-    "async_exit_stack==1.0.1",
-    "outcome==1.0.0",
-    "packaging==20.4",
-]
-
-
-test_requirements = [
-    "pytest==5.4.3",
-    "pytest-cov==2.10.0",
-    "pytest-xdist==1.32.0",
-    "pytest-trio==0.7.0",
-    "pytest-qt==3.3.0",
-    "pytest-rerunfailures==9.0",
-    "hypothesis==5.3.0",
-    "hypothesis-trio==0.5.0",
-    "trustme==0.6.0",
-    'winfspy[test]==0.8.2;platform_system=="Windows"',
-    # Fix botocore and sphinx conflicting requirements on docutils
-    "docutils>=0.12,<0.16",
-    # Documentation generation requirements
-    "sphinx==2.4.3",
-    "sphinx-intl==2.0.0",
-    "sphinx-rtd-theme==0.4.3",
-    "psutil==5.8.0",
-]
+def get_requirement_from_file(file):
+    with open(file) as requirements_txt:
+        return str(requirements_txt.read()).split("\n")
```

Comment:
no need for the `str` conversion

but encoding should be specified in the open instead (otherwise system default encoding might be used instead of utf8).

(and you can fix the lacking encoding option when reading README and HISTORY btw.)

Reply:
And parsing is wrong here: we can't have comments in the requirement.txt file

in theory we should use `pkg_resources.parse_requirements` to parse a requirements.txt file, however pkg_resources is a big module (slow to load) and depends of setuptools so it really seems like heavy artillery for the job

good thing is [pkg_resources.parse_requirements is simple enough](https://github.com/pypa/pip/blob/a800ae2a7d680c0ed5ba21476f722e6416a5ce0d/src/pip/_vendor/pkg_resources/__init__.py#L3075-L3094)

So I think we should just copy the parser code in `get_requirement_from_file`:
```python
def _maybe_add_as_continuation(current_line, previous_lines):
    try:
        last_line = previous_lines[-1]
    except IndexError:
        return False
    if last_line.endswith("\\"):
        previous_lines[-1]` = last_line[:-1] .strip() + current_line
        return True
    else:
        return False

lines = []
for line in requirements.iterlines():
    line = line.strip()
    if not line or line.startswith("#"):
        continue
    # Drop comments -- a hash without a space may be in a URL.
    line = line[:line.find(' #')]
    # If there is a line continuation, drop it, and append the next line.
    if not _maybe_add_as_continuation((line, lines):
        lines.append(line)
_maybe_add_as_continuation("", lines)  # In case file finish with \\ and no further line
```
Note I've changed continuation handling from the original code given:
1) it seems to me it wasn't handling multiple line continuation properly
2) the [removal of the continuation character)(https://github.com/pypa/pip/blob/a800ae2a7d680c0ed5ba21476f722e6416a5ce0d/src/pip/_vendor/pkg_resources/__init__.py#L3089) also remove the previous character, which imply it must be a space but this is [not what the spec says](https://pip.pypa.io/en/stable/reference/requirements-file-format/#line-continuations)

Reply:
> however pkg_resources is a big module (slow to load) and depends of setuptools so it really seems like heavy artillery for the job

I beg to differ :)
```python
In [1]: import setuptools

In [2]: %time import pkg_resources
CPU times: user 4 µs, sys: 0 ns, total: 4 µs
Wall time: 4.77 µs
```